server/aesdsocket.c

```diff
@@ -0,0 +1,301 @@
+
+	hints.ai_family = PF_INET;
+	hints.ai_socktype = SOCK_STREAM;
+	hints.ai_flags = AI_PASSIVE;
+	errorcode = getaddrinfo(NULL, port, &hints, &result);
+	if (errorcode != 0)
+	{
+		syslog(LOG_PERROR, "Error in gettng address suing getaddrinfo with error code %d\n", errno);
+		closelog();
+		exit(-1);
+	}
+
+	socket_fd = socket(result->ai_family, result->ai_socktype, 0);
+	if (socket_fd == -1)
+	{
+		syslog(LOG_PERROR, "socket creation failed with error code %d\n", errno);
+		closelog();
+		exit(-1);
+	}
+	syslog(LOG_DEBUG, "Socket Created with Socket Id %d\n", socket_fd);
+
+	int var_setsockopt = 1;
+	if (setsockopt(socket_fd, SOL_SOCKET, SO_REUSEADDR, &var_setsockopt,
+				   sizeof(int)) == -1)
+	{
+		syslog(LOG_PERROR, "Socket resuse address failed with error code %d\n", errno);
+		closelog();
+		exit(-1);
+	}
+
+	if (argc > 1 && strcmp(argv[1], "-d") == 0)
+	{
+		syslog(LOG_USER, "Daemonizing process\n");
+		daemonize();
+	}
+
+	errorcode = bind(socket_fd, result->ai_addr, result->ai_addrlen); // check what should be the sockaddr for it?
+	if (errorcode == -1)
+	{
+		close(socket_fd);
+		syslog(LOG_PERROR, "Socket bind failed with error code %d and closing server socket \n", errno);
+		closelog();
+		exit(-1);
+	}
+
+	syslog(LOG_DEBUG, "Bind Successful\n");
+	freeaddrinfo(result);
+
+	errorcode = listen(socket_fd, 10);
+	if (errorcode == -1)
+	{
+		close(socket_fd);
+		syslog(LOG_PERROR, "Listen failed with error code %d and closing server socket \n", errno);
+		closelog();
+		exit(-1);
+	}
+
+	struct sockaddr_in client_addr;
+	socklen_t client_addr_size = sizeof(client_addr);
+
+	while (1)
+	{
+		client_fd = accept(socket_fd, (struct sockaddr *)&client_addr, &client_addr_size);
+		if (client_fd == -1)
+		{
+			close(socket_fd);
+			syslog(LOG_PERROR, "connection failed with error code %d and closing server socket \n", errno);
+			closelog();
+		}
+
+		else
+		{
+			struct sockaddr_in *ip_addr_ptr = (struct sockaddr_in *)&client_addr;
+
+			inet_ntop(AF_INET, &ip_addr_ptr->sin_addr, ip_addr, INET_ADDRSTRLEN);
+
+			syslog(LOG_DEBUG, "Accepted connection from %s\n", ip_addr);
+
+			int fd;
+			fd = open(filename, O_RDWR | O_CREAT | O_APPEND, S_IRWXU | S_IRWXG | S_IRWXO);
+
+			if (fd == -1)
+			{
+				syslog(LOG_DEBUG, "Closed connection from %s\n", ip_addr);
+				close(client_fd);
+				syslog(LOG_PERROR, "Error in opening of the file %s and with error code %d\n", filename, errno);
+			}
+			else
+			{
+
+				syslog(LOG_DEBUG, "Recieve Started\n");
+
+				int bytes_rec;
+				bool rec_complete = false;
+				char *ptr = NULL;
+				while (rec_complete == false)
+				{
+
+					bytes_rec = recv(client_fd, file_array, FILE_SIZE, 0);
+					if (bytes_rec < 0)
+					{
+						syslog(LOG_DEBUG, "Closed connection from %s\n", ip_addr);
+						close(client_fd);
+						syslog(LOG_PERROR, "recieve unsuccessful with error code %d\n", errno);
+						closelog();
+					}
+					else
+					{
+						ssize_t result = write(fd, file_array, bytes_rec);
```

Comment:
@pranjalgupta1901 this implementation writes the file before the packet completes, which may cause problems with future assignments.